ls1intum/Artemis
# New User statistics charts

### Checklist

### Motivation and Context


We want to have more charts because charts are life. I am still figuring out if all those charts should be displayed as a bar chart. -> follow-up PR
Also, on Production, longer waiting time for the data seems to create weird looking charts

### Description

Added a couple of new charts:
- Logged-In users
- Exercises due
- Conducted exams
- Exam participations
- Exam registrations
- Active tutors
- Created result
- Created feedbacks

The charts display only zeros until data is loaded

### Steps for Testing


1. Log in to Artemis
2. Navigate to Server Administration -> User statistics
3. Make sure no initial error are thrown
4. Keep in mind the values for the important bars

Test 'Logged-In users' by first logging in with a test-account, checking the number and afterwards logging in with a not-test account. The number should increase by one if you did not already login with this account in the bar period.

Testing Exercise related stats
1. Create a exercise with due date in the future
2. participate in that exercise
3. assess that exercise, keep in mind the amount of feedbacks
4. Look into user statistics - corresponding bars in the charts should rise with the expected amount

Testing Exam related stats
1. Create an exam (its better to manually add the students, is relevant for 'Exam participations', the endDate of the exam is relevant for the 'conducted exam' chart
2. participate in that exam
3. Look into user statistics - corresponding bars in the charts should rise with the expected amount

### Screenshots


The new charts, with different spans:
<img width="1434" alt="Screenshot 2020-12-18 at 18 48 27" src="https://user-images.githubusercontent.com/44401560/102644957-fdc34c80-4161-11eb-902f-2864a3074b40.png">


<img width="1432" alt="Screenshot 2020-12-18 at 18 49 07" src="https://user-images.githubusercontent.com/44401560/102644980-0451c400-4162-11eb-8bb7-f3988e160945.png">


<img width="1435" alt="Screenshot 2020-12-18 at 18 49 32" src="https://user-images.githubusercontent.com/44401560/102644992-09af0e80-4162-11eb-88f1-56f045d85da7.png">
